Shared/Samples/Show viewshed from camera in scene/ShowViewshedFromCameraInSceneView.swift

```diff
@@ -0,0 +1,130 @@
+            analysisOverlays: [model.analysisOverlay]
+        )
+        .onAppear {
+            // Set up the initial viewshed when the view appears.
+            model.setInitialViewshed()
+        }
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                viewshedButton
+            }
+        }
+    }
+    
+    private var viewshedButton: some View {
+        Button("Viewshed from here") {
+            // Update viewshed based on current camera location when button is tapped.
+            model.updateViewshedFromCurrentCamera()
+        }
+        .buttonStyle(.borderedProminent)
+        .tint(.purple)
+        .padding(.horizontal)
+    }
+}
+
+private extension ShowViewshedFromCameraInSceneView {
+    @MainActor
+    @Observable
+    final class Model {
+        var currentCamera: Camera? = .initialCamera
+        /// An analysis overlay used to display the viewshed analysis visualization.
+        let analysisOverlay = AnalysisOverlay()
+        /// The viewshed which is updated by the camera.
+        @ObservationIgnored private var viewshed: LocationViewshed?
+        /// 3D Scene setup with imagery basemap, elevation, and mesh layer.
+        let scene: ArcGIS.Scene = {
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            scene.baseSurface.addElevationSource(
+                ArcGISTiledElevationSource(url: .elevation)
+            )
+            scene.addOperationalLayer(
+                IntegratedMeshLayer(url: .gironaMeshService)
+            )
+            return scene
+        }()
+        
+        /// Applies a viewshed at the initial camera position.
+        func setInitialViewshed() {
+            guard let camera = currentCamera else { return }
+            setupViewshed(using: camera)
+        }
+        
+        /// Updates the viewshed using the current camera position.
+        func updateViewshedFromCurrentCamera() {
+            guard let camera = currentCamera else { return }
+            viewshed?.update(from: camera)
+        }
+        
+        /// Applies the viewshed to the scene and sets the UI for analysis.
+        private func setupViewshed(using camera: Camera) {
+            // Create and configure the new viewshed.
+            let newViewshed = LocationViewshed(
+                camera: camera,
+                minDistance: 1.0,
+                maxDistance: 1000.0
+            )
+            // Set visual appearance of the viewshed.
+            Viewshed.visibleColor = UIColor.green.withAlphaComponent(0.5)
+            Viewshed.obstructedColor = UIColor.red.withAlphaComponent(0.5)
+            // Add the new viewshed to the overlay.
+            analysisOverlay.addAnalysis(newViewshed)
+            viewshed = newViewshed
+        }
+    }
+}
+
+private extension Camera {
+    
+    /// Camera that is 200m above ground over Girona, Spain,
+    /// looking slightly downward at the landscape.
+    static var initialCamera: Camera {
```

Comment:
No need to repeat type name.
```suggestion
    static var initial: Camera {
```

Reply:
But we don't need this static property. It is only being used in one spot so we can just create the camera on line 54.